frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx

```diff
@@ -71,17 +76,37 @@ export const generateClientKeySecret = (clientKey: string) => ({
   },
 });
 
-const generateCsiKmsConfigMap = (name: string, csiConfigData: KMSConfigMap) => ({
-  apiVersion: ConfigMapModel.apiVersion,
-  kind: ConfigMapModel.kind,
-  data: {
-    [`${name}`]: JSON.stringify(csiConfigData),
-  },
-  metadata: {
-    name: KMSConfigMapCSIName,
-    namespace: CEPH_STORAGE_NAMESPACE,
-  },
-});
+const convertCsiKeysToCamelCase = (csiConfigData: KMSConfigMap) => {
+  // ceph-csi supports only camelcase config keys for vault SA.
+  // And partially support uppser case for vault token.
+  // To avoid confusion, I am converting all vault config keys to camelcase.
+  const configData = csiConfigData as VaultSAConfigMap;
+  const res = Object.keys(configData).reduce(
+    (obj, key) => Object.assign(obj, { [KmsCsiConfigKeysMapping[key]]: configData[key] }),
+    {},
+  );
+  return res;
+};
+
+const generateCsiKmsConfigMap = (
+  name: string,
+  csiConfigData: KMSConfigMap,
+  convertToCamelCase = true,
+) => {
+  return {
+    apiVersion: ConfigMapModel.apiVersion,
+    kind: ConfigMapModel.kind,
+    data: {
+      [`${name}`]: JSON.stringify(
```

Comment:
Something to note. JSON `stringify` throws an error. So we need to validate the data to be a valid object.

Reply:
We have one such validation in UI, while displaying all the connection details as a dropdown. So, if there is any syntax error in the object (while parsing) it will be filtered out from the list and user will not be able to select those particular connection details.
Also, we are creating this object every time from our code, can there be any case when object can be invalid (just wondering could there be any case when this can happen) ??

Just my opinion, we can handle error to be on safer side though...

Reply:
Yes, we are already validating each field in UI